### ISSUE TITLE
Implement feature_importance as permutation and/or drop_feature based

### ISSUE THREAD

**Issue by andersbogsnes** [andersbogsnes/ml_tooling#59]
Sklearns standard implementation of randomforest feature importance relies on mean decrease in entropy, which, while fast can lead to inaccurate results, particularly in cases of differing scale:
see: https://www.stat.berkeley.edu/~breiman/RandomForests/cc_home.htm#varimp

There are two main alternatives: feature permutation and feature dropping, which both trade off speed for accuracy.

Feature Permutation
---
Feature permutation works by permuting each column in turn, scoring the model with the permuted features and comparing to the baseline score. The difference in baseline score compared to permuted score becomes the feature importance.

### Pros
- More accurate than mean decrease in entropy as calculate difference from random
- Doesn't require retraining the model, just a single training step and multiple scoring steps
- Can handle collinearity if we permute multiple features simultaneously
- Handles any kind of model, not just random forest
### Cons
- Slower than sklearn implementation
- Requires validation set for scoring

Feature dropping
---
Works similarly, except we drop each column from the dataset and train a new model, comparing to the baseline model

### Pros
- Most accurate version
- Handles any kind of model, not just random forest
- Can also handle collinearity by dropping multiple features

### Cons
- slooow! Need to train x + 1 models, where x is number of features

Both are worthwhile additions, and feature permutation should be default feature_importance, with the option of choosing feature dropping if desired.

Reference implementation: 
[rfpimp](https://github.com/parrt/random-forest-importances)
